src/psyclone/tests/domain/nemo/transformations/nemo_arrayrange2loop_trans_test.py

```diff
@@ -331,20 +332,115 @@ def test_apply_structure_of_arrays():
         "    enddo\n"
         "  enddo\n" in result)
 
-    # Case 2: SoA in the LHS is not yet supported
-    assert "mystruct%field2%field(:,:,:) = 0.0d0" in result
+    # Case 2: SoA in the LHS
+    array_ref = assignment2.lhs
+    trans.apply(array_ref.member.member.children[2])
+    trans.apply(array_ref.member.member.children[1])
+    trans.apply(array_ref.member.member.children[0])
+    result = fortran_writer(schedule)
+    assert (
+        "  do jk = 1, jpk, 1\n"
+        "    do jj = 1, jpj, 1\n"
+        "      do ji = 1, jpi, 1\n"
+        "        mystruct%field2%field(ji,jj,jk) = 0.0d0\n"
+        "      enddo\n"
+        "    enddo\n"
+        "  enddo\n" in result)
 
-    # Case 3: Nested SoA currently causes an InternalError
+    # Case 3: AoSoA in the LHS
```

Comment:
Isn't this case actually SoAoS or have I misunderstood the notation?

Reply:
You are right, its SoAoS